R/learner_mgcv_classif_gam.R

```diff
@@ -0,0 +1,190 @@
+#' @title Classification Generalized Additive Model Learner
+#' @author JazzyPierrot
+#' @name mlr_learners_classif.gam
+#'
+#' @description
+#' Generalized additive models.
+#' Calls [mgcv::gam] from package \CRANpkg{mgcv}.
+#'
+#' Multiclass classification is not implemented yet.
+#'
+#' A gam formula specific to the task at hand is required for the `formula`
+#' parameter (see example and `?mgcv::formula.gam`). Beware, if no formula is
+#' provided, a fallback formula is used that will make the gam behave like a
+#' glm (this behavior is required for the unit tests). Only features specified
+#' in the formula will be used, superseding columns with col_roles "feature"
+#' in the task.
+#'
+#' @template class_learner
+#' @templateVar id classif.gam
+#' @templateVar caller gam
+#'
+#' @references
+#'
+#' Wood, S.N. (2017) Generalized Additive Models: An Introduction with R (2nd edition). Chapman
+#' & Hall/ CRC, Boca Raton, Florida
+#'
+#' Key Reference on GAMs generally:
+#'
+#' Hastie (1993) in Chambers and Hastie (1993) Statistical Models in S. Chapman
+#' and Hall.
+#'
+#' Hastie and Tibshirani (1990) Generalized Additive Models. Chapman and Hall.
+#'
+#' @template seealso_learner
+#' @template example
+#'
+#' @examples
+#'
+#' # simple example
+#' t = mlr3::tsk("spam")
+#' l = mlr3::lrn("classif.gam")
+#' l$param_set$values = list(
+#'   formula = type ~ s(george) + s(charDollar) + s(edu) + ti(george, edu)
+#' )
+#' l$train(t)
+#' l$predict(t)
```

Comment:
```suggestion
#' l$model
```

I think it will be more informative to demonstrate the effect it has internally (and it means we can also avoid dealing with train/predict on same data)